01-shapefile-attributes.Rmd

```diff
@@ -75,182 +79,292 @@ CSV to Shapefile in R</a>
+lesson, you can skip this code chunk.
+
+```{r load-packages-data }
+#load packages
+library(rgdal)  #for vector work; sp package should always load with rgdal. 
+library (raster)   #for metadata/attributes- vectors or rasters
+
+#set working directory to data folder
+#setwd("pathToDirHere")
 
 #Import a polygon shapefile 
-aoiBoundary <- readOGR("boundaryFiles/HARV/", "HarClip_UTMZ18")
+aoiBoundary_HARV <- readOGR("boundaryFiles/HARV/", "HarClip_UTMZ18")
 
 #Import a line shapefile
-lines <- readOGR( "boundaryFiles/HARV/",layer = "HARV_roads")
+lines_HARV <- readOGR( "boundaryFiles/HARV/", "HARV_roads")
 
 #Import a point shapefile 
-point <- readOGR("boundaryFiles/HARV/", "HARVtower_UTM18N")
+point_HARV <- readOGR("boundaryFiles/HARV/", "HARVtower_UTM18N")
 
 ```
 
-##Querying shapefile metadata 
-Recall from the previous lesson that shapefile metadata can include a `class`, 
-a `features` count, an `extent`, and a `coordinate reference system` (crs). 
-
-Metadata can be extracted individually using the `class()`, `length()`, 
-`extent()`, `crs()`, `R` commands. 
+#Query Shapefile Metadata 
+Shapefile metadata can include object `class`, a features count (`length`), the
+object `extent`, and the object's coordinate reference system (`CRS`). To query
+the shapefile metadata the code is s
 
 ```{r view-shapefile-metadata }
-
 #view class
-class(x = aoiBoundary)
-class(x = lines)
-class(x = point)
+class(x = aoiBoundary_HARV)
+class(x = lines_HARV)
+class(x = point_HARV)
 
+# x= isn't actually needed; it just specifies which object
 #view features count
-length(x = aoiBoundary)
-length(x = lines)
-length(x = point)
+length(aoiBoundary_HARV)
+length(lines_HARV)
+length(point_HARV)
 
 #view crs - note - this only works with the raster package loaded
-crs(x = aoiBoundary)
-crs(x = lines)
-crs(x = point)
+crs(aoiBoundary_HARV)
+crs(lines_HARV)
+crs(point_HARV)
 
-#view extent
-extent(x = aoiBoundary)
-extent(x = lines)
-extent(x = point)
+#view extent- note - this only works with the raster package loaded
+extent(aoiBoundary_HARV)
+extent(lines_HARV)
+extent(point_HARV)
 
 ```
 
-#Querying Shapefile Attributes
+When we type in the full file name we see the same three types of metadata with
+some additional attribute information.  
+
+```{r view-shapefile-metadata2 }
+
+aoiBoundary_HARV
+lines_HARV
+point_HARV
+
+```
+
+#Query Shapefile Attributes
 Shapefiles often contain an associated database or spreadsheet of values called
 `attributes` that describe the vector features in the shapefile. You can think
 of this like a spreadsheet with rows and columns. Each column in the spreadsheet
-is an individual `Attribute` that describes an object. Shapefile `attributes` 
-include measurements that correspond to the geometry of the shapefile features.
+is an individual `attribute` that describes an object (sometimes called
+variables). Shapefile `attributes` include measurements that correspond to the
+geometry of the shapefile features.
```

Comment:
Ok i think i wrote about this in the previous lesson? we might need to double check for redundancy after these are re built.